apis/management.cattle.io/v3/schema_types.go

```diff
@@ -51,6 +52,7 @@ type Field struct {
 	ValidChars   string   `json:"validChars,omitempty"`
 	InvalidChars string   `json:"invalidChars,omitempty"`
 	Description  string   `json:"description,omitempty"`
+	DynamicField bool     `json:"dynamicField,omitempty"`
```

Comment:
What is DynamicField used for?

Reply:
When dynamic schemas are created or updated the schema created from the dynamic schema is merged with the existing schema.  The problem with this is that if a field is removed from the dynamic schema that field is not removed from the existing schema.  This makes it impossible to remove a dynamic field from a schema.

If dynamic field is set to true, then that field is removed from the existing schema before it is merged with the new schema.

The reason it has to exist, and isn't always just assumed to be true implicitly, is because the dynamic schema is actually marshaled to json, and then unmarshaled to the schema type. This only works because schema and dynamic schema are functionally identical despite being completely different types.  So while dynamic field will always be set to `True` on dynamic schema, it will be set to false on `Schema`.